desktop/ui/src/main/java/org/datacleaner/windows/AbstractDatastoreDialog.java

```diff
@@ -120,16 +120,20 @@ public void actionPerformed(ActionEvent e) {
         _datastoreNameTextField.getDocument().addDocumentListener(new DCDocumentListener() {
             @Override
             protected void onChange(DocumentEvent event) {
-                validateAndUpdate();
+                validateAndUpdateInternal();
             }
         });
     }
-    
+
     protected void validateAndUpdate() {
+        validateAndUpdateInternal();
+    }
+
+    private void validateAndUpdateInternal() {
```

Comment:
I don't think so... Validation is just validation, it's the "AndUpdate" part of validateAndUpdate that is the problem :)